iauploader.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+"""
+Retrieve and disseminate files and metadata to Internet Archive
+"""
+
+import logging
+import sys
+from internetarchive import upload
+from io import BytesIO
+from os import environ
+from uploader import Uploader
+
+
+class IAUploader(Uploader):
+    """Dissemination logic for Internet Archive"""
+
+    def upload_to_platform(self):
+        """Upload work in required format to Internet Archive"""
+
+        # Identifier and filenames TBD: use paperback ISBN for now
+        filename = self.get_pb_isbn()
+        # Metadata file format TBD: use CSV for now
+        metadata_bytes = self.get_formatted_metadata('csv::thoth')
+        pdf_bytes = self.get_pdf_bytes()
+
+        # Convert Thoth work metadata into Internet Archive format
+        ia_metadata = self.parse_metadata()
+
+        responses = upload(
+            identifier=filename,
+            files={
+                '{}.pdf'.format(filename): BytesIO(pdf_bytes),
+                '{}.csv'.format(filename): BytesIO(metadata_bytes),
+            },
+            metadata=ia_metadata,
+            access_key=environ.get('ia_s3_access'),
+            secret_key=environ.get('ia_s3_secret'),
+        )
+
+        for response in responses:
+            if response.status_code != 200:
+                logging.error(
+                    'Error uploading to Internet Archive: {}'.format(response.text))
+                sys.exit(1)
+
+        logging.info(
+            'Successfully uploaded to Internet Archive at archive.org/details/{}'.format(filename))
+
+    def parse_metadata(self):
+        """Convert work metadata into Internet Archive format"""
+        work_metadata = self.metadata.get('data').get('work')
+        ia_metadata = {
+            'title': work_metadata.get('title'),
+            'publisher': self.get_publisher_name(),
+            'creator': work_metadata.get('copyrightHolder'),
```

Comment:
I'm not entirely sure the creator should be the copyright holder. Could this be perhaps a concatenation of the main contributions?

Reply:
Agreed - this was intended as a placeholder, although I've now got a handle on repeatable fields (e.g. creator) and added some more. As discussed, we may well move this logic to the export server anyway (creating a new JSON output).